special-pages/pages/new-tab/app/customizer/components/VisibilityMenu.js

```diff
@@ -16,49 +16,55 @@ import { useTypedTranslation } from '../../types.js';
-                                                stroke-width="1.5"
-                                                stroke-linecap="round"
-                                                stroke-linejoin="round"
-                                            />
-                                        </svg>
-                                    )}
-                                </span>
-                                <span className={styles.svg}>
-                                    {row.icon === 'shield' && <DuckFoot />}
-                                    {row.icon === 'star' && <Shield />}
-                                </span>
-                                <span>{row.title ?? row.id}</span>
-                            </label>
-                        </li>
-                    );
-                })}
-            </ul>
-        </div>
+        <ul className={styles.list}>
+            {rows.map((row) => {
+                return (
+                    <li key={row.id}>
+                        <label className={styles.menuItemLabel} htmlFor={MENU_ID + row.id}>
+                            <input
+                                type="checkbox"
+                                checked={row.visibility === 'visible'}
+                                onChange={() => row.toggle?.(row.id)}
+                                id={MENU_ID + row.id}
+                                class={styles.checkbox}
+                            />
+                            <span aria-hidden={true} className={styles.checkboxIcon}>
+                                {row.visibility === 'visible' && (
+                                    <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+                                        <path
+                                            d="M3.5 9L6 11.5L12.5 5"
+                                            stroke="white"
+                                            stroke-width="1.5"
+                                            stroke-linecap="round"
+                                            stroke-linejoin="round"
+                                        />
+                                    </svg>
+                                )}
+                            </span>
+                            <span className={styles.svg}>
+                                {row.icon === 'shield' && <DuckFoot />}
+                                {row.icon === 'star' && <Shield />}
+                            </span>
+                            <span>{row.title ?? row.id}</span>
+                        </label>
+                    </li>
+                );
+            })}
+        </ul>
     );
 }
+
+export function Heading() {
+    const { t } = useTypedTranslation();
+    return <h2 className="sr-only">{t('widgets_visibility_menu_title')}</h2>;
+}
+
+export function VisibilityMenuPopover({ children }) {
+    return <div className={styles.dropdownInner}>{children}</div>;
+}
```

Comment:
just extracted for better composition - this allows us to separate the popover menu styling from when it's just a list